plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -132,6 +132,7 @@ nonZeroArg :: (Integer -> Integer -> Integer) -> Integer -> Integer -> Evaluatio
 nonZeroArg _ _ 0 = EvaluationFailure
 nonZeroArg f x y = EvaluationSuccess $ f x y
 
+-- We need to inline local functions, so that more efficient Core is generated.
```

Comment:
So if we add more builtins, we should remember to do that?

Reply:
If they use local definitions, then yes. But I'm going to try changing this whole machinery completely. Although that rule about local functions is probably going to stay anyway.